src/bridge/mod.rs

```diff
@@ -44,6 +47,84 @@ fn neovim_instance() -> Result<NeovimInstance> {
     }
 }
 
+/// Takes the --cmd or -c argument and returns the command to be executed.
+#[cfg(target_os = "macos")]
+fn handle_command_arg(position: usize, args: Vec<String>) -> String {
+    args.get(position + 1).cloned().unwrap_or_default()
+}
+
+/// Takes the valid path argument and returns the startup directory.
+#[cfg(target_os = "macos")]
+fn handle_command_arg_as_path_or_default(args: &mut Vec<String>) -> String {
+    args.retain(|arg| is_valid_path(arg));
+
+    let path = args.first().cloned().unwrap_or_default();
+    let startup_directory = get_startup_directory(&path);
+
+    format!(
+        "if g:neovide_ntty == v:true | chdir {} | endif",
+        startup_directory
+    )
+}
+
+/// The function `setup_tty_startup_directory` sets up the startup directory for
+/// a new TTY Neovim session on macOS platform, such as Finder, Dock, or even
+/// external programs like Neohub.
+///
+/// Any nvim --cmd or -c argument is handled as a command to be executed.
+///
+/// Conditions:
+///
+/// - Is not a TTY session.
+/// - Argument is a directory, it becomes the startup directory.
+/// - Argument is a file, its parent directory becomes the startup directory.
+/// - Neither directory nor file, $HOME is used.
+#[cfg(target_os = "macos")]
```

Comment:
just confirm to me that the default startup path in windows or linux (if you have it) aren't $HOME launching from a non tty, then I make it platform agnostic.

Reply:
It looks like my comment has jumped around, it was meant for the `handle_command_arg_as_path_or_default`, which should be possible to support on all platforms even if the default startup directory isn't changed otherwise.

Reply:
that was it, makes sense @fredizzimo 

done!